File: infrastructure/lib/infrastructure-table-stack.ts
import {aws_dynamodb as dynamodb, CfnOutput, RemovalPolicy, Stack, StackProps} from 'aws-cdk-lib';
import {Construct} from 'constructs';

export class InfrastructureTableStack extends Stack {
    constructor(scope: Construct, id: string, props?: StackProps) {
        super(scope, id, props);

        const tableName = 'Products-JVM-Example';
        const productsTable = new dynamodb.Table(this, id, {
            billingMode: dynamodb.BillingMode.PROVISIONED,
            writeCapacity: 8,
            readCapacity: 8,
            removalPolicy: RemovalPolicy.DESTROY,
            partitionKey: {name: 'id', type: dynamodb.AttributeType.STRING},
            pointInTimeRecovery: false,
            tableName: tableName,
        });

        new CfnOutput(this, `${tableName}-table-arn`, {
            value: productsTable.tableArn,
            description: `The arn of the ${tableName} table`,
            exportName: `${tableName}TableArn`,
        });
    }
}

File: infrastructure/src/main/kotlin/nl/vintik/sample/cdk/InfrastructureTableStack.kt
package nl.vintik.sample.cdk

import software.amazon.awscdk.CfnOutput
import software.amazon.awscdk.CfnOutputProps
import software.amazon.awscdk.RemovalPolicy
import software.amazon.awscdk.Stack
import software.amazon.awscdk.StackProps
import software.amazon.awscdk.services.dynamodb.Attribute
import software.amazon.awscdk.services.dynamodb.AttributeType
import software.amazon.awscdk.services.dynamodb.BillingMode
import software.amazon.awscdk.services.dynamodb.Table
import software.constructs.Construct

class InfrastructureTableStack(scope: Construct, id: String, props: StackProps) : Stack(scope, id, props) {
    init {
        val tableName = "Products-JVM-Example"
        val productsTable = Table.Builder.create(this, tableName)
            .tableName(tableName)
            .partitionKey(
                Attribute.builder()
                    .type(AttributeType.STRING)
                    .name("id")
                    .build()
            )
            .removalPolicy(RemovalPolicy.DESTROY)
            .pointInTimeRecovery(false)
            .billingMode(BillingMode.PAY_PER_REQUEST)
            .build()

        CfnOutput(
            this, "${tableName}-table-arn",
            CfnOutputProps.builder()
                .value(productsTable.tableArn)
                .description("The arn of the $tableName table")
                .exportName("${tableName}TableArn").build()
        )
    }
}